stytch/core/http/client.py

```diff
@@ -26,48 +35,68 @@ def __init__(self, project_id: str, secret: str) -> None:
         super().__init__(project_id, secret)
         self.auth = requests.auth.HTTPBasicAuth(project_id, secret)
 
-    def get(self, url: str, params: Optional[Dict[str, Any]]) -> requests.Response:
-        return requests.get(url, params=params, headers=self.headers, auth=self.auth)
+    @classmethod
+    def _response_from_request(cls, r: requests.Response) -> ResponseWithJson:
+        try:
+            resp_json = r.json()
+        except Exception:
+            resp_json = {}
+        return ResponseWithJson(response=r, json=resp_json)
+
+    def get(self, url: str, params: Optional[Dict[str, Any]]) -> ResponseWithJson:
+        resp = requests.get(url, params=params, headers=self.headers, auth=self.auth)
+        return self._response_from_request(resp)
 
-    def post(self, url: str, json: Optional[Dict[str, Any]]) -> requests.Response:
-        return requests.post(url, json=json, headers=self.headers, auth=self.auth)
+    def post(self, url: str, json: Optional[Dict[str, Any]]) -> ResponseWithJson:
+        resp = requests.post(url, json=json, headers=self.headers, auth=self.auth)
+        return self._response_from_request(resp)
 
-    def put(self, url: str, json: Optional[Dict[str, Any]]) -> requests.Response:
-        return requests.put(url, json=json, headers=self.headers, auth=self.auth)
+    def put(self, url: str, json: Optional[Dict[str, Any]]) -> ResponseWithJson:
+        resp = requests.put(url, json=json, headers=self.headers, auth=self.auth)
+        return self._response_from_request(resp)
 
-    def delete(self, url: str) -> requests.Response:
-        return requests.delete(url, headers=self.headers, auth=self.auth)
+    def delete(self, url: str) -> ResponseWithJson:
+        resp = requests.delete(url, headers=self.headers, auth=self.auth)
+        return self._response_from_request(resp)
 
 
 class AsyncClient(ClientBase):
     def __init__(self, project_id: str, secret: str) -> None:
         super().__init__(project_id, secret)
         self.auth = aiohttp.BasicAuth(project_id, secret)
 
-    async def get(
-        self, url: str, params: Optional[Dict[str, Any]]
-    ) -> aiohttp.ClientResponse:
+    @classmethod
+    async def _response_from_request(
+        cls, r: aiohttp.ClientResponse
+    ) -> ResponseWithJson:
+        try:
+            resp_json = await r.json()
+        except Exception:
+            resp_json = {}
+        return ResponseWithJson(response=r, json=resp_json)
```

Comment:
[pebble] I feel a little weird about dropping this exception without logging or storing it. Based on what I've seen so far, I'd expect this to eventually turn into a `raise StytchError` with an "unknown" type, which I think is safe. But I think it'll be tough to debug parse or read errors like this.

Can we store the exception in `ResponseWithJson`? Is it better to throw a special "client error"?